engine/internal/autoscaler/autoscaler.go

```diff
@@ -182,12 +192,24 @@ func (s *scaler) scale(ctx context.Context) error {
 	} else {
 		s.lastTransitToZero = time.Time{}
 	}
-	return s.scaleTo(ctx, &sts, cappedReplicas)
+
+	specReplicas := float64(ptr.Deref(sts.Spec.Replicas, 0))
+
+	if newReplicas < specReplicas {
+		maxScaleDown := math.Floor(float64(replicas) * s.config.MaxScaleDownRate)
```

Comment:
Checking my understanding. If`MaxScaleDownRate` is 0.2, and the current replica size is 10, we scale down the stateful set to 2 replicas when it is most aggressive. 

When the number of replica becomes 2, `maxScaleDown` is `floor(2*0.2)= 0`. If the min replica is 0, the target replica size becomes 0.